src/MetricsBuilder.php

```diff
@@ -244,6 +247,47 @@ private function appendCredentialsMetric(
         }
     }
 
+    /**
+     * Appends the account_id_endpoint_mode metrics based on
+     * the value resolved.
+     *
+     * @param string $accountIdEndpointMode
+     *
+     * @return void
+     */
+    private function appendAccountIdEndpointMode(
+        string $accountIdEndpointMode
+    ): void
+    {
+        if (empty($accountIdEndpointMode)) {
+            return;
+        }
+
+        if ($accountIdEndpointMode === 'preferred') {
+            $this->append(MetricsBuilder::ACCOUNT_ID_MODE_PREFERRED);
```

Comment:
calling from `self` context for all of these

Reply:
Done.